Makefile

```diff
@@ -79,6 +79,20 @@ include $(ROOT_DIR)/flight/targets/*/target-defs.mk
 # OpenPilot GCS build configuration (debug | release)
 GCS_BUILD_CONF ?= debug
 
+# And the flight build configuration (debug | default | release)
+export FLIGHT_BUILD_CONF ?= default
+
+ifeq ($(FLIGHT_BUILD_CONF), debug)
+export DEBUG:=YES
+else ifeq ($(FLIGHT_BUILD_CONF), release)
+export DEBUG:=NO
+else ifeq ($(FLIGHT_BUILD_CONF), default)
+export DEBUG:=NO
```

Comment:
This line will have the constant side effect of exporting ````DEBUG:=NO```` even for non-flight builds.  That might be very confusing if that variable ever becomes important for gcs, or androidgcs or uavo builds.

If the user happens to have ````FLIGHT_BUILD_CONF = debug```` in their environment, this code will constantly assert ````DEBUG:=YES```` even for non-flight builds which may be even more confusing.

It might make more sense to:
* move the setting of DEBUG out of this makefile and into the flight makefiles
* keep the validation of FLIGHT_BUILD_CONF values in this file but move it down under the "Check that environmental variables are sane" heading with the other validation code.